### PR TITLE
Add RPC to force round timeout

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Add RPC to force round timeout

### Tested

CI

### CODE REVIEW

File: consensus/istanbul/backend/api.go [hunk 1]
```diff
@@ -153,6 +153,12 @@ func (api *API) GetCurrentRoundState() (*core.RoundStateSummary, error) {
 	return api.istanbul.core.CurrentRoundState().Summary(), nil
 }
 
+// GetCurrentRoundState retrieves the current IBFT RoundState
+func (api *API) ForceRoundChange() (bool, error) {
+	api.istanbul.core.ForceRoundChange()
```
Comment (timmoreton) [inline consensus/istanbul/backend/api.go]: check if engine is started here?

File: consensus/istanbul/core/handler.go [hunk 2]
```diff
@@ -211,17 +211,17 @@ func (c *core) handleCheckedMsg(msg *istanbul.Message, src istanbul.Validator) e
 	return errInvalidMessage
 }
 
-func (c *core) handleTimeoutAndMoveToNextRound(desiredView *istanbul.View) error {
-	logger := c.newLogger("func", "handleTimeoutAndMoveToNextRound", "set_at_seq", desiredView.Sequence, "set_at_desiredRound", desiredView.Round)
+func (c *core) handleTimeoutAndMoveToNextRound(timedOutView *istanbul.View) error {
+	logger := c.newLogger("func", "handleTimeoutAndMoveToNextRound", "set_at_seq", timedOutView.Sequence, "set_at_desiredRound", timedOutView.Round)
```
Comment (timmoreton) [inline consensus/istanbul/core/handler.go]: maybe `"timed_out_seq"` etc too?

File: consensus/istanbul/backend/api.go [hunk 3]
```diff
@@ -150,9 +150,21 @@ func (api *API) GetValEnodeTable() (map[string]*vet.ValEnodeEntryInfo, error) {
 
 // GetCurrentRoundState retrieves the current IBFT RoundState
 func (api *API) GetCurrentRoundState() (*core.RoundStateSummary, error) {
+	if api.istanbul.coreStarted {
+		return nil, istanbul.ErrStoppedEngine
+	}
```
Comment (lucasege) [inline consensus/istanbul/backend/api.go]: Is this correct? Seems to be like there should be a `not` here, as in if the core is *not* started, return `ErrStoppedEngine`.
Maybe I'm missing something, but if so maybe a comment would be helpful.

File: consensus/istanbul/core/types.go [hunk 4]
```diff
@@ -25,11 +25,15 @@ import (
...
 	Start() error
 	Stop() error
+	// CurrentView returns the current view or nil if none
 	CurrentView() *istanbul.View
+	// CurrentRoundState returns the current roundState or nil if none
 	CurrentRoundState() RoundState
 	SetAddress(common.Address)
 	// Validator -> CommittedSeal from Parent Block
 	ParentCommits() MessageSet
+	// ForceRoundChange will force round change to the current deriredRound + 1
```
Comment (lucasege) [inline consensus/istanbul/core/types.go]: nit: `deriredRound` -> `desiredRound`

File: consensus/istanbul/backend/api.go [hunk 5]
```diff
@@ -150,9 +150,21 @@ func (api *API) GetValEnodeTable() (map[string]*vet.ValEnodeEntryInfo, error) {
...
 	return api.istanbul.core.CurrentRoundState().Summary(), nil
 }
 
+// GetCurrentRoundState retrieves the current IBFT RoundState
```
Comment (nategraf) [inline consensus/istanbul/backend/api.go]: Comment should be updated

File: consensus/istanbul/backend/api.go (hunk 3 above)
Comment (mcortesi) [reply consensus/istanbul/backend/api.go]: heh... you're right!

File: consensus/istanbul/backend/api.go [hunk 6]
```diff
@@ -150,9 +150,21 @@ func (api *API) GetValEnodeTable() (map[string]*vet.ValEnodeEntryInfo, error) {
...
+func (api *API) ForceRoundChange() (bool, error) {
+	if !api.istanbul.coreStarted {
+		return false, istanbul.ErrStoppedEngine
+	}
+	api.istanbul.core.ForceRoundChange()
```
Comment (nategraf) [inline consensus/istanbul/backend/api.go]: What happens when this is called while the engine is already waiting for the next round?

Comment (timmoreton) [reply consensus/istanbul/backend/api.go]: it will be as though that timeout got sped up, and we then move to the next round.

**Pull request merged**